src/Esprima/ParserExtensions.cs

```diff
@@ -28,6 +28,13 @@ static ParserExtensions()
             Scanner.TryGetInternedPunctuator(source);
     }
 
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal static string Slice(this string source, int start, int end, ref StringPool stringPool)
+    {
+        var sourceSpan = source.AsSpan(start, end - start);
+        return TryGetInternedString(sourceSpan) ?? stringPool.GetOrCreate(sourceSpan);
+    }
+
     public static string Slice(this string source, int start, int end)
```

Comment:
I think this is no longer used and can probably be removed? seems that also `CharToString` should probably by internal, maybe even `CharCodeAt`?

Reply:
I agree. I think we can make the whole `ParserExtensions` class internal for now.